src/battle_main.c

```diff
@@ -3081,6 +3087,56 @@ void FaintClearSetData(void)
             gBattleMons[i].status2 &= ~(STATUS2_INFATUATED_WITH(gActiveBattler));
         if ((gBattleMons[i].status2 & STATUS2_WRAPPED) && *(gBattleStruct->wrappedBy + i) == gActiveBattler)
             gBattleMons[i].status2 &= ~(STATUS2_WRAPPED);
+
+        // If the fainted mon was holding another Pokemon in Sky Drop, release the mon and clear Sky Drop data
+        if (gActiveBattler == gBattleStruct->skyDropTargets[0] && i == gBattleStruct->skyDropTargets[1])
+        {
+            gBattleStruct->skyDropTargets[0] = 0xFF;
+            gBattleStruct->skyDropTargets[1] = 0xFF;
+            
+            gStatuses3[i] &= ~(STATUS3_SKY_DROPPED | STATUS3_ON_AIR);
+            gSprites[gBattlerSpriteIds[i]].invisible = FALSE;
+            
+            // If the target was sky dropped in the middle of using Outrage/Petal Dance/Thrash,
+            // confuse them upon release and print "confused via fatigue" message and animation.
+            if (gBattleMons[i].status2 & STATUS2_LOCK_CONFUSE)
```

Comment:
the status3 and status2 flags are cleared at the top of this function, so this will always be false.